code-challenges.test.js

```diff
@@ -1,51 +1,155 @@
+    expect(endTally(votes2)).toEqual(-31)
+  })
+})
+//------ TEST CASE FAILING DUE TO FUNCTION NOT BEING DEFINED-----
+// FAIL  ./code-challenges.test.js
+// endTally
+//   ✕ Returns the upvotes minus the down votes (10 ms)
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+// ● endTally › Returns the upvotes minus the down votes
+
+//   ReferenceError: endTally is not defined
 
 // b) Create the function that makes the test pass.
+//------PSEUDO CODE-------
+//---input: object
+//---- putput: integer 
+// create a function called endTally that takes in a given object
+// declare an empty that takes the values from the Object using the Object.values(object) built in method
+// apply a negative to the down votes values and then add the two indexs together in the array to get the end tally of the votes.
+
+const endTally = (object) => {
+  const newArray = Object.values(object)
+  return newArray[0] + (newArray[1] * -1) 
+}
+
+//------ TEST CASE PASSING WITH FUNCION------
+// PASS  ./code-challenges.test.js
+// endTally
+//   ✓ Returns the upvotes minus the down votes (4 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+describe("noDuplicates", ()=> {
+  it("Combines arrays and removes duplicates", ()=> {
+      const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+      const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+      // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+    expect(noDuplicates(dataArray1,dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
+//-------FAILED TEST DUE TO FUNCTION NOT DEFINED------
+// FAIL  ./code-challenges.test.js
+// noDuplicates
+//   ✕ Combines arrays and removes duplicates (1 ms)
+// ● noDuplicates › Combines arrays and removes duplicates
+//   ReferenceError: noDuplicates is not defined
 
 // b) Create the function that makes the test pass.
+//-------Pseudo Code-------
+// INPUT: two seperate arrays
+// OUTPUT: single array (with no duplicates)
+// Create a function called noDuplicates with parameters of array1 & array 2
+// create a new array called setArray
+// make the new array equal to the two argument arrays concated and apply the <...new Set>
+// spread operator on them which will remove duplicates
+// return the setArray
+
```

Comment:
I love this approach! The new Set is the one removing duplicates, what is the spread operator doing?